lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -372,9 +372,17 @@ public function removeElement($element)
                 return $this->collection->removeElement($element);
             }
 
+            if ($this->em->getUnitOfWork()->getEntityState($element) === UnitOfWork::STATE_NEW) {
+                return true;
+            }
+
             $persister = $this->em->getUnitOfWork()->getCollectionPersister($this->association);
 
-            return $persister->removeElement($this, $element);
+            if ($persister->removeElement($this, $element) === true) {
```

Comment:
It's kind of risky to leave this outside of a transaction. That creates confusion and would be lovely if we could avoid it IMHO

Reply:
@lcobucci I agree, but its too late now. We can't move this behavior out of a transaction without completly removing the extra lazy from `remoteElement`.

Reply:
Even if we track the IDs of removed elements and filter them from the select query?

Reply:
Nerver mind, you're right about it being too late.